sonar-batch/src/main/java/org/sonar/batch/issue/DeprecatedIssueWrapper.java

```diff
@@ -0,0 +1,178 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.batch.issue;
+
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import org.sonar.api.batch.fs.TextRange;
+import org.sonar.api.batch.rule.Severity;
+import org.sonar.api.batch.sensor.issue.internal.DefaultIssue;
+import org.sonar.api.issue.Issue;
+import org.sonar.api.issue.IssueComment;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.Duration;
+
+public class DeprecatedIssueWrapper implements Issue {
+
+  private final DefaultIssue newIssue;
+
+  public DeprecatedIssueWrapper(DefaultIssue newIssue) {
+    this.newIssue = newIssue;
+  }
+
+  public DefaultIssue wrapped() {
+    return newIssue;
+  }
+
+  @Override
+  public String key() {
+    return newIssue.key();
+  }
+
+  @Override
+  public String componentKey() {
+    return null;
+  }
+
+  @Override
+  public RuleKey ruleKey() {
+    return newIssue.ruleKey();
+  }
+
+  @Override
+  public String language() {
+    return null;
+  }
+
+  @Override
+  public String severity() {
+    Severity overridenSeverity = newIssue.overridenSeverity();
+    return overridenSeverity != null ? overridenSeverity.name() : null;
+  }
+
+  @Override
+  public String message() {
+    return newIssue.locations().get(0).message();
+  }
+
+  @Override
+  public Integer line() {
+    TextRange textRange = newIssue.locations().get(0).textRange();
+    return textRange != null ? textRange.start().line() : null;
+  }
+
+  @Override
+  public Double effortToFix() {
+    return newIssue.effortToFix();
+  }
+
+  @Override
+  public String status() {
+    return null;
+  }
+
+  @Override
+  public String resolution() {
+    return null;
+  }
+
+  @Override
+  public String reporter() {
+    return null;
+  }
+
+  @Override
+  public String assignee() {
+    return null;
+  }
+
+  @Override
+  public Date creationDate() {
+    return null;
+  }
+
+  @Override
+  public Date updateDate() {
+    return null;
+  }
+
+  @Override
+  public Date closeDate() {
+    return null;
+  }
+
+  @Override
+  public String attribute(String key) {
+    return null;
+  }
+
+  @Override
+  public Map<String, String> attributes() {
+    return null;
+  }
+
+  @Override
+  public String authorLogin() {
+    return null;
+  }
+
+  @Override
+  public String actionPlanKey() {
+    return null;
+  }
+
+  @Override
+  public List<IssueComment> comments() {
+    return null;
+  }
+
+  @Override
+  public boolean isNew() {
+    return false;
+  }
+
+  @Override
+  public Duration debt() {
+    return null;
+  }
+
+  @Override
+  public String projectKey() {
+    return null;
+  }
+
+  @Override
+  public String projectUuid() {
+    return null;
+  }
+
+  @Override
+  public String componentUuid() {
+    return null;
+  }
+
+  @Override
+  public Collection<String> tags() {
+    return null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1168)